src/main/java/com/revature/services/PersonLogicImpl.java

```diff
@@ -1,34 +1,90 @@
 package com.revature.services;
 
+
+
+
 import java.util.List;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotWritableException;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Isolation;
 import org.springframework.transaction.annotation.Transactional;
 
 import com.revature.domain.Person;
 import com.revature.repositories.PersonRepository;
 
+
 @Service
 @Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
 public class PersonLogicImpl implements PersonLogic {
 
+
 	@Autowired
-	private PersonRepository personRepository;
+	private PersonRepository dao;
+	
+
 	
 	@Override
 	public List<Person> getPersonByFirstName(String firstName) {
 		
-		return personRepository.findByFirstName(firstName);
+		return  dao.findByFirstName(firstName);
 
 	}
 
 	@Override
 	public List<Person> getPersonByLastName(String lastName) {
 		
-		return personRepository.findByLastName(lastName);
+		return dao.findByLastName(lastName);
 	}
 	
 
+
+	@Override
+	public List<Person> getPersonsByPersonRole(int personRole) {
+		
+		return dao.findByPersonRole(personRole);
+	
+	}
+
+	@Override
+	public List<Person> getAllTrainees() {
+		
+		return dao.findByPersonRole(1);
+
+	}
+
+	@Override
+	public List<Person> getAllTrainers() {
+		
+		return dao.findByPersonRole(2);
+
+	}
+
+	@Override
+	public Person getPersonById(int id)  {
+		
+		Person p = null;
+		
+		try {
+			 
+			p = dao.getOne(id);
+			System.out.println("This is my P: "+p);
+			 
+		 } catch (Exception e) {		
+			 
+			 p = new Person("","",0);
+			 p.setId(0);
+			 System.out.println("setting P to new person");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS106)